tests/components/element-editor/conditional-rendering/conditional-rendering-child.spec.tsx

```diff
@@ -0,0 +1,138 @@
+            },
+            {
+                builderContext: {
+                    focusedElement: { element: MOCK_ELEMENT },
+                },
+            }
+        );
+
+        expect(screen.getByText("Select")).toBeInTheDocument();
+        expect(screen.getByText("Equals")).toBeInTheDocument();
+        expect(screen.getByPlaceholderText("Set value")).toBeInTheDocument();
+    });
+
+    it("should fire the delete function on clicking the bin icon", () => {
+        renderComponent(
+            {
+                onDelete: mockDelete,
+                options: mockOptions,
+                onChange: mockOnChange,
+                value: mockValue,
+            },
+            {
+                builderContext: {
+                    focusedElement: { element: MOCK_ELEMENT },
+                },
+            }
+        );
+        const deleteButton = screen.getByTestId("delete-button");
+        fireEvent.click(deleteButton);
+        expect(mockDelete).toBeCalled();
+    });
+
+    it("should fire onChange when there is a change in the input fields", () => {
+        renderComponent(
+            {
+                onDelete: mockDelete,
+                options: mockOptions,
+                onChange: mockOnChange,
+                value: mockValue,
+            },
+            {
+                builderContext: {
+                    focusedElement: { element: MOCK_ELEMENT },
+                },
+            }
+        );
+        const getValueField = screen.getByPlaceholderText("Set value");
+        fireEvent.focus(getValueField);
+        fireEvent.change(getValueField, {
+            target: { value: "mockValue" },
+        });
+        expect(mockOnChange).toBeCalled();
+    });
+});
+
+interface IConditionalRenderingChildOptions {
+    onDelete?: () => void;
+    options?: IOptions[];
+    onChange?: (newValue: any) => void;
+    value?: IConditionalRendering;
+}
+
+const MyTestComponent = ({
+    onDelete,
+    options,
+    onChange,
+    value,
+}: IConditionalRenderingChildOptions) => {
+    return (
+        <ConditionalRenderingChild
+            onDelete={onDelete}
+            options={options}
+            onChange={onChange}
+            value={value}
+        />
+    );
+};
+
+const renderComponent = (
+    conditionalRenderingChildOptions: IConditionalRenderingChildOptions = {},
+    overrideOptions?: TestHelper.RenderOptions
+) => {
+    return render(
+        TestHelper.withProviders(
+            overrideOptions,
+            <MyTestComponent {...conditionalRenderingChildOptions} />
+        )
+    );
+};
+
+// =============================================================================
+// MOCKS
+// =============================================================================
+const mockOptions = [{ label: "mockLabel1", id: "mockId1" }];
+const mockDelete = jest.fn();
+const mockOnChange = jest.fn();
+const mockValue = {
+    ["mock123"]: { fieldKey: "", comparator: "Equals", value: "" },
```

Comment:
```suggestion
    mock123: { fieldKey: "", comparator: "Equals", value: "" },
```

Reply:
amended this accordiong to the change in object shape :)